content/guides/higher-order-functions.adoc

```diff
@@ -0,0 +1,184 @@
+
+[source, clojure]
+----
+(defn double-+
+    [a b]
+    (* 2 (+ a b)))
+----
+
+[source, clojure]
+----
+(defn double-*
+    [a b]
+    (* 2 (* a b)))
+----
+
+These functions share a common pattern.
+They only differ in name and the function used in the computation of `a` and `b`.
+In general, the pattern looks like this.
+
+[source, clojure]
+----
+(defn double-<f>
+    [a b]
+    (* 2 (f a b)))
+----
+
+We can generalize our `double-` function by passing `f` in as an argument.
+
+[source, clojure]
+----
+(defn double-op
+    [f a b]
+    (* 2 (f a b)))
+----
+
+We can use this to express the concept of doubling the result of an operation rather than having to write functions that perform specific doubled operations individually.
+
+== Anonymous Functions
+
+An anonymous function is a function without a name.
+In Clojure these can be defined in two ways, `fn` and the literal `#(...)`.
+Creating a function with `defn` immediately binds it to a name, `fn` just creates a function.
+
+Lets go back to the example of the bands.
+We wrote a function called `rock-band?` to determine whether or not our band was a rock band.
+This is a one-off operation, we're not going to use it anywhere else in our code.
+We can save ourselves some keystrokes by using an anonymous function.
+
+[source, clojure]
+----
+(def rock-bands
+    (filter
+        (fn [band] (= :rock (:genre band)))
+        bands))
+----
+
+Even more concisely, using the function literal, we can define `rock-bands` like so.
+
+[source, clojure]
+----
+(def rock-bands (filter #(= :rock (:genre %)) bands))
+----
+
+The function literal supports multiple arguments via `%1`, `%2`, and so on.
+
+[source, clojure]
+----
+#(println %1 %2 %3)
+----
+
+If you're writing an anonymous function, the literal syntax is nice because it's so compact.
+However, beyond a few arguments, the syntax can become difficult to read.
+In that case, using `fn` may be more appropriate.
+
+=== Functions Returning Functions and Closures
+
+Our first function will be called `adder`.
+It will take a number, `x`, as it's only argument and return a function.
+The function returned by `adder` will also take a single number, `a`, as it's argument and return `x + a`.
+
+[source, clojure]
+----
+(defn adder [x]
+  (fn [a] (+ x a)))
+
+(def add-five (adder 5))
+
+(add-five 100)
+;; => 105
+----
+
+The returned function form `adder` is a closure.
+This means, it can access all of the variables that we're in scope when the function was created.
+`add-five` has access to `x` even though it is outside of the `adder` function definition.
+
+=== Filter
+
+Filtering is a common operation in computer programming.
+Take this set of animals
+
+[source, clojure]
+----
+(def pets [
+    {:name "Fluffykins" :type :cat}
+    {:name "Sparky" :type :dog}
+    {:name "Tibby" :type :dog}
+    {:name "Al" :type :fish}
+    {:name "Victor" :type :bear}
+])
+----
+
+We want to filter out the dogs because we're writing enterprise grade software.
```

Comment:
Here, the "out" implies removal to me, but you want to keep the dogs. Maybe replace "filter out" with "keep only".